coralproject/talk
# [CORL-2370] Add archiving copy to stream/admin

## What does this PR do?

- Injects the `archivingEnabled` and `autoArchiveOlderThanMs` into the `staticConfig` and from there into the `Local` relay store for both admin and stream
- If archiving is enabled, shows the following throughout stream/admin
    - `My Profile > my comments` will indicate that the comments shown are only those that are new enough to not be archived based on the archiving age threshold
    - Dashboard labels will show that they are relative to the archiving threshold and no longer for "all time"
    - User history dashboard will indicate that the comments shown are only those that are new enough to not be archived based on the archiving age threshold

## What changes to the GraphQL/Database Schema does this PR introduce?

None

## How do I test this PR?

- Enable archiving by setting `MONGODB_ARCHIVE_URI` in your `.env` file
    - I use a second mongo instance using this command: 
     ```
     docker run -d -p 27018:27017 --restart always --name mongo-archive mongo:4.2
     ```
    - I then use the connection string for that second Mongo I spun up for archiving in my `.env` file:
     ```
     MONGODB_ARCHIVE_URI=mongodb://127.0.0.1:27018/coral
     ```
    - The reason I spin up another Mongo is because archiving is meant to be used with archived comments living in a different cluster (mongo instance) than the live comments. This docker setup matches production most closely for local testing.
- Build and run using `npm run build:development` and `npm run start:development`
- Use an external commenting/iframe host (i.e. the `multisite-test-bed`)
    - This is because the `npm run watch` tooling does not properly update the local state and staticConfig injection due to its configuration
- Check that `My Profile > my comments`, the dashboard, and the user history all comments tab show the appropriate archiving specific notifications and labels
    - Further details on what that should be are listed above in `What does this PR do?`
 
## How do we deploy this PR?

No special considerations should be needed.
